apollo-router/src/plugins/reporting.rs

```diff
@@ -254,11 +265,29 @@ impl Plugin for Reporting {
             }
             tracing::debug!("terminating spaceport loop");
         });
+
+        let mut opentracing_layer = None;
+        if let Some(opentracing_conf) = &configuration.opentracing {
+            opentracing_layer = OpenTracingLayer::new(opentracing_conf.clone())?.into();
+        }
+
         Ok(Reporting {
             config: configuration,
             tx,
+            opentracing_layer,
         })
     }
+
+    fn subgraph_service(
+        &mut self,
+        _name: &str,
+        service: BoxService<SubgraphRequest, SubgraphResponse, BoxError>,
+    ) -> BoxService<SubgraphRequest, SubgraphResponse, BoxError> {
+        match &self.opentracing_layer {
+            Some(opentracing_layer) => opentracing_layer.layer(service).boxed(),
```

Comment:
Nit: consider using https://docs.rs/tower/latest/tower/util/fn.option_layer.html

Reply:
I assume you would like something like this: 
```
ServiceBuilder::new()
            .layer(option_layer(self.opentracing_layer))
            .service(service)
            .boxed()
``` 

but it requires me to clone self.opentracing_layer which is not what I want. Correct me if I'm wrong

Reply:
Fair enough :)